challenge.rb

```diff
@@ -1,26 +1,27 @@
 
 def capitalize_each_string(input)
-  #implement your solution here
+  input.collect(&:capitalize)
 end
 
 def fetch_the_dog(input)
-  #implement your solution here
+  # In preference to turning the result of .detect or .find into an Array.
+  input.find_all { |animal| animal == "dog" }
 end
 
 def no_dogs_allowed(input)
-  #implement your solution here
+  input.reject { |animal| animal == "dog" }
 end
 
 def count_the_animals(input)
-  #implement your solution here
+  input.count
 end
 
 def fetch_the_first_two(input)
-  #implement your solution here
+  input.first(2)
 end
 
 def fetch_CD_animals(input)
-  #implement your solution here
+  input.grep(/^[cCdD]/)
```

Comment:
Rather than listing both cases of each letter, I think it is slightly better to use the case insensitive flag on the regular expression. `/^[cd]/i`

Also, Ruby regular expressions are multline by default. So `^` matches the beginning of any line. If you want to restrict the match to the beginning of the string, use `\A`.